codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -214,9 +218,41 @@ public ChannelFuture sendFlowControlledFrame(ChannelHandlerContext ctx, Http2Str
     }
 
     @Override
-    public ChannelFuture lastFlowControlledFrameSent(Http2Stream stream) {
-        FlowState state = state(stream);
-        return state != null ? state.lastNewFrame() : null;
+    public ChannelFuture sendHeaders(ChannelHandlerContext ctx, Http2Stream stream, Http2Headers headers,
+            int streamDependency, short weight, boolean exclusive, int padding,
+            boolean endOfStream, ChannelPromise promise) {
+      boolean isFirst = !sentFrame;
```

Comment:
Theres definitely some tension here. If we want the encoder to be the only thing that knows about stream state then it either needs to do the first HEADERs write itself or we have two interfaces on the flow-controller so that it's told which one to do. 

I'll do the former and clarify the interface to the controller that it's only responsible for trailing/intermediate headers